processor/k8sattributesprocessor/internal/kube/fake_informer.go

```diff
@@ -145,7 +145,7 @@ func (f *NoOpInformer) AddEventHandler(handler cache.ResourceEventHandler) (cach
 	return f.AddEventHandlerWithResyncPeriod(handler, time.Second)
 }
 func (f *NoOpInformer) AddEventHandlerWithResyncPeriod(handler cache.ResourceEventHandler, resyncPeriod time.Duration) (cache.ResourceEventHandlerRegistration, error) {
-	return handler, nil
+	return nil, nil
```

Comment:
Is this change intended?

Reply:
Yes, this is a fake informer that is only used for testing from what I can tell. This was created to satisfy the `[SharedInformer](https://pkg.go.dev/k8s.io/client-go/tools/cache@v0.27.1#SharedInformer)` interface

Reply:
I asked myself that question also but could not find any reason why we could not just return nil here. Unit tests pass and I am hoping that e2e test show the same. I'm fixing the go mod issue as we speak.